<launch>

  <arg name="rviz" default="true" />
  <arg name="preprocess" default="true" />

  <node name="ximea" pkg="ximea_camera" type="ximea_camera_node" output="screen" >
    <param name="frame_rate" type="int" value="10" />
    <param name="bandwidth_mbps" type="int" value="8000" />
    <rosparam param="camera_param_file_paths" subst_value="true">[$(find ximea_camera)/config/custom_ximea.yaml]</rosparam>
  </node>

<!-- <node name = "calibration" pkg= "camera_calibration" type ="cameracalibrator.py" output = "screen">
    <param name = "size" value ="8x6"/>
    <param name = "square" value = "0.0255"/>
    <remap from="image" to="/camera/image_raw" />
    <remap from="camera" to="/camera" />
</node> -->


<!-- <node ns = "camera" name="proc" pkg="image_proc" type="image_proc"/> -->
<!-- <node ns = "preprocess" name="debayering" pkg="nodelet" type="nodelet" args = "standalone image_proc/debayer"  output="screen">
    <remap from="image_raw" to="/camera/image_raw" />
</node> -->
<group if="$(arg preprocess)" ns = "preprocess">
  <node name="resize" pkg="nodelet" type="nodelet" args = "standalone image_proc/resize">  output="screen"
      <remap from="image_raw" to="/camera/image_raw" /> #image_mono
      <remap from="camera_info" to="/camera/camera_info" /> #image_mono
      <param name="use_scale" type = "bool" value="true"/>
      <param name="scale_height" type = "double" value="0.5"/>
      <param name="scale_width" type = "double" value="0.5"/>
  </node>

  <!-- either do the full image recitification before orbslam2 or within orbslam2 -->
  <node name="rectification" pkg="nodelet" type="nodelet" args = "standalone image_proc/rectify">  output="screen"
      <remap from="image_mono" to="resize/image" />
      <remap from="camera_info" to="resize/camera_info" />
  </node>
</group>
  <group if="$(arg rviz)">
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find ximea_camera)/config/vis.rviz" />
  </group>

</launch>
